examples/crm/src/dataProvider.ts

```diff
@@ -11,20 +16,45 @@ const TASK_MARKED_AS_UNDONE = 'TASK_MARKED_AS_UNDONE';
 const TASK_DONE_NOT_CHANGED = 'TASK_DONE_NOT_CHANGED';
 let taskUpdateType = TASK_DONE_NOT_CHANGED;
 
+const beforeContactUpsert = async (
+    params: CreateParams<any> | UpdateParams<any>,
+    dataProvider: DataProvider
+) => {
+    const { data } = params;
+    const avatarUrl = await getAvatarUrl(data);
+    data.avatar = avatarUrl || null;
+
+    if (!data.company_id) {
+        return params;
+    }
+
+    const { data: company } = await dataProvider.getOne('companies', {
+        id: data.company_id,
+    });
+
+    if (!company) {
+        return params;
+    }
+
+    data.company_name = company.name;
+    return params;
+};
+
 export const dataProvider = withLifecycleCallbacks(baseDataProvider, [
     {
         resource: 'contacts',
-        beforeCreate: async params => {
-            const { data } = params;
-            const avatarUrl = await getAvatarUrl(data);
-            data.avatar = avatarUrl || null;
-            return params;
+        beforeCreate: async (params, dataProvider) => {
+            return beforeContactUpsert(params, dataProvider);
         },
-        beforeUpdate: async params => {
-            const { data } = params;
-            const avatarUrl = await getAvatarUrl(data);
-            data.avatar = avatarUrl || null;
-            return params;
+        beforeUpdate: async (
+            params: UpdateParams<any>,
+            dataProvider: DataProvider
+        ) => {
+            const result = await beforeContactUpsert(params, dataProvider);
+            return {
```

Comment:
why do you modify the return object from beforeContactUpsert here but not in beforeCreate? Is it reeally necessary?